GitObjectDb/Models/LazyLink.cs

```diff
@@ -64,7 +64,7 @@ public TLink Link
             {
                 if (Parent == null)
                 {
-                    throw new NotSupportedException($"Parent is not attached to {nameof(LazyLink<TLink>)}.");
+                    throw new GitObjectDbException($"Parent is not attached to {nameof(LazyLink<TLink>)}.");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the exception throwing from this property getter, or refactor the property into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2372)